lib/mongo/collection.rb

```diff
@@ -204,17 +216,37 @@ def write_concern_with_session(session)
       wc
     end
 
-    # Provides a new collection with either a new read preference or new write concern
-    # merged over the existing read preference / write concern.
+    # Provides a new collection with either a new read preference, new read
+    # concern or new write concern merged over the existing read preference /
+    # read concern / write concern.
```

Comment:
As per RUBY-3084, the `CHANGEABLE_OPTIONS` contains both `read` and `read_concern`. The `read` option corresponds to the `read_preference` options. Now, this method has only documented the ability to change the read preference and write concern, while allowing the changing of read preference, **read concern**, and write concern.

There are two options here: 
- document that `with` also allows changing read concern, which is what's done here
- change `with` to not allow the changing of read concern as per RUBY-3084.

This PR implements the former. I can also move this snippet to its own PR if that would be better.

Reply:
https://jira.mongodb.org/browse/RUBY-3086

Reply:
I made the ticket before reading this PR, I see that you've done several of the changes listed there in this PR already.